### ISSUE TITLE
When fragment is the same type is omitted

### ISSUE THREAD

**Issue by kgtkr** [dotansimha/graphql-code-generator#1434]
**Describe the bug**
When fragment is the same type is omitted
But __typename does not work properly

**To Reproduce**
1. Use '...' as schema
2. Use '...' as documents
```
fragment x on X{
    ...a
    ...b
}

fragment a on A{
    p
}

fragment b on B{
    p
}
```
3. Run the following command: `gql-gen ...`

**Expected behavior**
```ts
export namespace X {
  export type Fragment =
    | A.Fragment
    | B.Fragment
}
```

*** Output ***
```ts
export namespace X {
  export type Fragment =
    | A.Fragment
}
```

**Schema/Documents**
If possible and relevant, please provide your GraphQL schema and GraphQL documents.

**Environment:**
 - OS:
 - Codegen:
 - Node: 

**Additional context**

**Comment by dotansimha** [comment]
I think we fixed it during the refactor (still WIP): https://github.com/dotansimha/graphql-code-generator/pull/1353 

To make sure, can you please provide a schema matches your types? are you using interfaces or unions for the fragment matching? 
Also, can you please provide an example of the output you are expecting?

**Comment by kgtkr** [comment]
expecting:
```ts
export namespace X {
  export type Fragment =
    | A.Fragment
    | B.Fragment
}
```

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍
